packages/babel-plugin-emotion/src/index.js

```diff
@@ -450,16 +453,31 @@ export default function(babel: Babel) {
         exit(path: BabelPath, state: EmotionBabelPluginPass) {
           if (state.staticRules.length !== 0) {
             const toWrite = state.staticRules.join('\n').trim()
-            const filenameArr = path.hub.file.opts.filename.split('.')
-            filenameArr.pop()
-            filenameArr.push('emotion', 'css')
-            const cssFilename = filenameArr.join('.')
+            let cssFilename = path.hub.file.opts.sourceFileName
+
+            if (state.outputDir) {
+              cssFilename = nodePath.join(state.outputDir, cssFilename)
+            }
+
+            const cssFilenameArr = cssFilename.split('.')
+            // remove the extension
+            cssFilenameArr.pop()
+            // add emotion.css as an extension
+            cssFilenameArr.push('emotion.css')
+
+            // use absolute path so resolving can't go wrong
```

Comment:
We can't rely on absolute paths. There are so many times when it would cause problems, for example, distributing an npm package, the imports wouldn't work since they would point to paths that wouldn't exist on a consumers computer. Please use [`path.relative`](https://nodejs.org/api/path.html#path_path_relative_from_to) to get relative paths instead.